### DISCUSSION TITLE
Collection item owner

### DISCUSSION THREAD

**Discussion by louiswol94** [directus/directus#3567] [Archive]
Good day,

I would like to find out what is the best way to achieve the following:

We have a collection called "projects". When anyone creates a new "project" we want a dropdown field with all the Directus Users, then the chosen user in the dropdown is the only user that will be able to make "updates" to this project after it is created. We will call this field the "project owner" for example. We should update their permissions accordingly.

Thanks in advance.

**Comment by benhaynes** [comment]
You'll need to create permissions that:

* Allow anyone to create
* Update is limited to where `projects.owner` (the dropdown) is set to `$CURRENT_USER`

You should be able to figure it out from there!

**Reply by louiswol94** [reply]
Hi Ben,

Thanks for the response. I am quite new to Directus, so I would need to ask a few more questions to make sure I am doing this the right way.

When you say I need to: Update is limited to where projects.owner (the dropdown) is set to $CURRENT_USER

**Does this mean I need to:**

1) Modify the "settings" on the "Collection & Fields" in the Directus Admin area? (Cannot see such a setting for my use case currently)
2) Or do I need to do this in a filter `item.update.projects:before`? (Then programatically check if 'project owner' is 'current user'. How do I then stop the save and throw a custom error message inside this filter?)

Thanks!

**Reply by louiswol94** [reply]
If it is option 2, I imagine something like this should work:

```
'item.update.projects:before' => function (\Directus\Hook\Payload $payload) {
            $project_owner = '1'; // will get this value dynamically - '1' is just an example
            $container = \Directus\Application\Application::getInstance()->getContainer();
            $acl = $container->get('acl');
            $current_user_id = $acl->getUserId();
            
            if ( $project_owner !== $current_user_id ) {
                throw new Exception\RuntimeException(
                    'Only the project owner can make changes to this project'
                );
            }
},
```

**Reply by benhaynes** [reply]
Neither! You would do it in Settings > Permissions... setting the role CRUD permissions appropriately. We're planning on adding an article about this process soon!!

**Reply by louiswol94** [reply]
Hi Ben. I can see in the Settings where you set the CRUD operations, but I do not see where you can allow them to only update specific items in the collection where they have been selected as the project owner. Remember 'Project Owner' is just a dropdown field we added on the 'projects collection'. So anyone can create a project, but only the selected user from that dropdown field can edit this specific project. I might just be missing what you are trying to explain to me though. The above code snippet is working for now at least. I am looking forward to the blog post :-D

**Reply by rijkvanzanten** [reply]
@louiswol94 I think the confusion here comes from the fact that @benhaynes described the feature in v9, while your hook suggests to me that you're using v8 🙂 

v9 has way more granular permission controls which make your use case possible. v8 doesn't have that, so you're hook would be the way to go 👍🏻
